src/query/catalog/src/plan/partition.rs

```diff
@@ -89,21 +89,22 @@ impl Partitions {
     }
 
     pub fn reshuffle(&self, executors: Vec<String>) -> Result<HashMap<String, Partitions>> {
-        let mut executors_sort = executors;
-        executors_sort.sort();
+        let mut executors_sorted = executors;
+        executors_sorted.sort();
 
-        let executor_nums = executors_sort.len();
+        let num_executors = executors_sorted.len();
         let partitions = match self.kind {
             PartitionsShuffleKind::Seq => self.partitions.clone(),
             PartitionsShuffleKind::Mod => {
                 // Sort by hash%executor_nums.
-                let mut parts = self.partitions.clone();
-                parts.sort_by(|a, b| {
-                    let hl = a.hash() % executor_nums as u64;
-                    let hr = b.hash() % executor_nums as u64;
-                    hl.cmp(&hr)
-                });
-                parts
+                let mut parts = self
```

Comment:
@BohuTANG why do we need to `sort by hash%executor_nums.`

Reply:
It will make a part always send to the same node, so we can re-use the cache on the node.